### PR TITLE
Add scaffolding for request invite RPC

### PR DESCRIPTION
**Pull request by cjb**
@chrisnojima Here's an early look at the functional request invite code, in case you want to pull it in.

### CODE REVIEW

**Comment by chrisnojima** [comment]
:+1:

**Comment by cjb** [comment]
@keybase/react-hackers Added styling and RPC to this PR:

Zeplin:

screen shot 2016-03-15 at 3 02 10 pm

screen shot 2016-03-15 at 3 02 28 pm

GIF:

reqinvite

File: shared/actions/signup/index.js [hunk 1]
```diff
@@ -48,6 +54,54 @@ export function checkInviteCode (inviteCode: string): TypedAsyncAction<CheckInvi
...
+    // Returns an error string if not valid
+    const emailError = isValidEmail(email)
+    const nameError = isValidName(name)
+    if (emailError || nameError || !email || !name) {
+      dispatch({
+        type: Constants.requestInvite,
+        error: true,
+        payload: {emailError, nameError, email, name}
+      })
+      resolve()
```
Comment (chrisnojima) [inline shared/actions/signup/index.js]: is this a resolve and not a reject?

File: shared/actions/signup/index.js [hunk 2]
```diff
@@ -56,13 +110,21 @@ function hasSpaces (s: string): boolean {
   return s.indexOf(' ') !== -1
 }
 
-// Returns an error string if not valid
-function isValidCommon (thing: ?string): ?string {
+function hasAtSign (s: string): boolean {
```
Comment (chrisnojima) [inline shared/actions/signup/index.js]: all this stuff really should be in go...

File: shared/login/signup/request-invite.render.desktop.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  render () {
+    const submitRequestInvite = () => {
+      this.props.onRequestInvite(this.state.email, this.state.name)
+    }
+
+    return (
+      <Container onBack={this.props.onBack} style={styles.container}>
+        <Text style={styles.header} type='Header'>Request an invite code</Text>
+        <Icon style={styles.icon} type='invite-code-m'/>
+        <Input
```
Comment (chrisnojima) [inline shared/login/signup/request-invite.render.desktop.js]: should set autoFocus

File: shared/login/signup/request-invite.render.desktop.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+          style={styles.input}
+          floatingLabelText='Your email address'
+          value={this.state.email || ''}
+          errorText={this.props.emailErrorText}
+          onChange={event => this.setState({email: event.target.value})}
+        />
+        <Input
+          style={styles.input}
+          floatingLabelText='Your name'
+          value={this.state.name || ''}
```
Comment (chrisnojima) [inline shared/login/signup/request-invite.render.desktop.js]: @MarcoPolo why do we flow type this thing to need a '' ?

**Comment by chrisnojima** [comment]
:+1: % c

File: shared/login/signup/request-invite.render.desktop.js (hunk 4 above)
Comment (MarcoPolo) [inline shared/login/signup/request-invite.render.desktop.js]: That's what you specified ;) 
https://github.com/keybase/client/blame/master/shared/common-adapters/input.js.flow#L18

You defined this type as an optional key, but if it is there it should be a string.

If you don't want that you can say I want an optional key and if it is there it is either a string or null.
That would look like: `value?: ?string`.

File: shared/actions/signup/index.js (hunk 1 above)
Comment (cjb) [inline shared/actions/signup/index.js]: thanks, switched to `reject(err)`

File: shared/login/signup/request-invite.render.desktop.js (hunk 4 above)
Comment (chrisnojima) [inline shared/login/signup/request-invite.render.desktop.js]: yeah let's change this, its annoying :)

**Pull request merged**